references/classification/train.py

```diff
@@ -165,10 +166,21 @@ def main(args):
     train_dir = os.path.join(args.data_path, 'train')
     val_dir = os.path.join(args.data_path, 'val')
     dataset, dataset_test, train_sampler, test_sampler = load_data(train_dir, val_dir, args)
+
+    collate_fn = None
+    num_classes = len(dataset.classes)
```

Comment:
not for now, but this exposes a limitation of our current datasets, which is that we don't consistently enforce a way of querying the number of classes in a dataset. The dataset refactoring work from @pmeier will address this

Reply:
With #4432, you will be able to do 

```python
info = torchvision.datasets.info(name)
info.categories
```

where `categories` is a list of strings in which the index corresponds to the label.